### PR TITLE
ExtractTensorPatches augmentation

### PR DESCRIPTION
**Pull request by isaaccorley**
#### Changes

Previously `kornia.contrib.ExtractTensorPatches` is just a `nn.Module`. This means when it's used inside a container like `AugmentationSequential` and the input has an `image` and a `mask`, only the `image` is transformed. To get around this I've refactored `ExtractTensorPatches` to a geometric augmentation.

The ideal case would be that if I'm training using a cropping augmentation like `K.RandomCrop(256)` and I want to evaluate on my larger images in the val or test sets with non-overlapping patches of the same size, one could use this to do this evaluation.

Fixes # (issue)

#### Type of change
- [ ] 📚  Documentation Update
- [ ] 🧪 Tests Cases
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [] 🔬 New feature (non-breaking change which adds functionality)
- [X] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] 📝 This change requires a documentation update

#### Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

**Review by edgarriba — COMMENTED** [review]
thanks for the patch. In terms of code organization, we want that specific Augmentation module to live in the kornia.augmentations to avoid polluting the core library. I suggest to have somewhere in kornia.augmentations a clone of the ExtractTensorPatches (maybe with a slightly different name to avoid globals conflicts). /cc @johnnv1 @shijianjian

**Comment by isaaccorley** [comment]
Thanks for the feedback! I'll work on refactoring!

**Comment by edgarriba** [comment]
@isaaccorley we are releasing by the end of the week. Do you think we can make it happen before that ?

**Comment by johnnv1** [comment]
hey @isaaccorley, will you continue working on this?

**Comment by isaaccorley** [comment]
@johnnv1 Yes I will have time in the near future but if you want to jump in and finish it feel free to!

**Comment by johnnv1** [comment]
as part of https://github.com/kornia/kornia/discussions/2719 plan, we have deleted the `master` in favor of `main`, if anyone has an interest in keep/restart working on this, let me know to help you.
